wrappers/services.go

```diff
@@ -277,10 +277,27 @@ func AddSnapServices(s *snap.Info, inter interacter) (err error) {
 		}
 
 		svcName := app.ServiceName()
-		if err := sysd.Enable(svcName); err != nil {
-			return err
+
+		// if this service is in the list of services to be left disabled, don't
+		// enable it
+		shouldEnable := true
+		for _, disabledSvc := range disabledSvcs {
+			// sanity check that all apps in the disabledSvcs list are in fact
+			// services
+			if !disabledSvc.IsService() {
+				return fmt.Errorf("snap app %s is not a service, cannot disable", disabledSvc.Name)
```

Comment:
This should be done earlier, probably before this line https://github.com/snapcore/snapd/pull/7437/files#diff-08088787360fb3ca74a0aed4c6103b22R207

Also, AFAIK it's perfectly legal if an app that was a service in the previous revision of the snap (as in previously installed, not exactly the n-1 revision), becomes a regular app in the current revision. Perhaps, we should log it, but not throw an error. Unless, those apps are filtered out earlier.

Reply:
hmm that's an interesting point, it is possible for services to become apps, given this I think you're right we shouldn't fail here then

Reply:
what I ended up doing here is to just log messages about previously disabled services that are now apps and previously disabled services that are missing from the snap, but the larger change will now check for both of those cases when handling that before providing it here, so in practice I don't think this codepath will be hit, but still good to be defensive here and log a notice if we do somehow hit it.